src/components/FilterModal/FilterModal.jsx

```diff
@@ -1,25 +1,51 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { Button } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
 import Modal from '../Modal/Modal';
 import CheckBoxList from '../CheckBoxList/CheckBoxList';
+import { getFilterSetter, getFilterResetter } from '../../filters/freaksFilter';
 
-function FilterModal({ title, isOpen, onClose, keywords }) {
+function FilterModal({ title, isOpen, onClose, keywords, setOpenModal }) {
   const [ checkedState, setCheckedState ] = useState([]);
+  const queuedFilters = useRef([]);
+  const dispatch = useDispatch();
+
+  const onChangeCb = (result) => {
+    setCheckedState(result);
+    queuedFilters.current = [ ...result ];
+  };
+
+  const onClickCb = () => {
+    const setFilter = getFilterSetter(title);
+    dispatch(setFilter(queuedFilters.current));
+    setOpenModal(false);
+  };
+
+  const resetModal = () => {
+    setCheckedState([]);
+    queuedFilters.current = [];
+    dispatch(getFilterResetter(title)());
+    setOpenModal(false);
+  };
 
   const getHeader = () => (
     <Button
       className="app-button -gray"
       type="button"
       variant="default"
-      onClick={ () => setCheckedState([]) }
+      onClick={ () => resetModal() }
```

Comment:
Here, you can just pass the reference to the function, as you don't need to add any extra arguments

```suggestion
      onClick={resetModal}
```

Reply:
77cec239e4d873200a72cb654497e48efe57c585